server/sonar-server/src/main/java/org/sonar/server/setting/ws/SetAction.java

```diff
@@ -319,10 +321,17 @@ private static SetRequest toWsRequest(Request request) {
   private static Map<String, String> readOneFieldValues(String json, String key) {
     Type type = new TypeToken<Map<String, String>>() {
     }.getType();
+    Gson gson = GsonHelper.create();
     try {
-      return (Map<String, String>) GsonHelper.create().fromJson(json, type);
-    } catch (JsonParseException e) {
-      throw new BadRequestException(String.format("Invalid JSON '%s' for setting '%s'", json, key));
+      return (Map<String, String>) gson.fromJson(json, type);
+    } catch (JsonSyntaxException e) {
+      try {
```

Comment:
Is this try/catch really useful ? Why displaying a different error if the JSON is not a map ?